File: frontend/src/components/Menu.js
import React, { useEffect, useState } from "react";

function Menu() {
  const [menuItems, setMenuItems] = useState([]);

  useEffect(() => {
    // Здесь делаем запрос к backend'у
    fetch("http://localhost:8080/menu")
      .then((response) => response.json())
      .then((data) => {
        setMenuItems(data);
      })
      .catch((error) => {
        console.error("Error fetching menu:", error);
      });
  }, []); // Пустой массив зависимостей означает, что запрос будет сделан только один раз при монтировании компонента

  return (
    <div>
      <h1>Menu</h1>
      <ul>
        {menuItems.map((item) => (
          <li key={item.id}>
            <h3>{item.name}</h3>
            <p>{item.description}</p>
            <p>{item.price} USD</p>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default Menu;

File: internal/handlers/menu.go
package handlers

import (
	"encoding/json"
	"log"
	"net/http"
	"restaurant-management/internal/services"
)

func GetMenuItems(w http.ResponseWriter, r *http.Request) {
	items, err := services.FetchMenuItems()
	if err != nil {
		http.Error(w, "Failed to fetch menu items", http.StatusInternalServerError)
		return
	}

	// Парсинг поля Nutrition для каждого элемента
	for i := range items {
		items[i].ParseNutrition(items[i].Nutrition.(string))
	}

	w.Header().Set("Content-Type", "application/json")
	if err := json.NewEncoder(w).Encode(items); err != nil {
		log.Printf("Error encoding response: %v", err)
		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
	}
}

// package handlers

// import (
// 	"context"
// 	"encoding/json"
// 	"net/http"
// 	"restaurant-management/internal/services"
// )

// type MenuItem struct {
// 	ID          int     `json:"id"`
// 	Name        string  `json:"name"`
// 	Description string  `json:"description"`
// 	Nutrition   string  `json:"nutrition"`
// 	Price       float64 `json:"price"`
// }

// func GetMenuItems(w http.ResponseWriter, r *http.Request) {
// 	rows, err := services.DB.Query(context.Background(), "SELECT id, name, description, nutrition, price FROM menu_items")
// 	if err != nil {
// 		http.Error(w, "Failed to fetch menu items", http.StatusInternalServerError)
// 		return
// 	}
// 	defer rows.Close()

// 	var menuItems []MenuItem

// 	for rows.Next() {
// 		var item MenuItem
// 		err := rows.Scan(&item.ID, &item.Name, &item.Description, &item.Nutrition, &item.Price)
// 		if err != nil {
// 			http.Error(w, "Error scanning menu item", http.StatusInternalServerError)
// 			return
// 		}
// 		menuItems = append(menuItems, item)
// 	}

// 	w.Header().Set("Content-Type", "application/json")
// 	json.NewEncoder(w).Encode(menuItems)
// }
